Does not work with `poetry@1.1.0`

All types generate this error:

```
E     main:1: error: Cannot find implementation or library stub for module named 'returns.example' (diff)
E     main:1: note: See https://mypy.readthedocs.io/en/latest/running_mypy.html#missing-imports (diff)
E     main:3: note: Revealed type is 'Any'          (diff)
```

Related https://github.com/dry-python/returns/pull/659
Related https://github.com/HypothesisWorks/hypothesis/issues/2635

This is a critical bug.

--- a/pytest_mypy_plugins/collect.py
+++ b/pytest_mypy_plugins/collect.py
@@ -18,7 +18,7 @@ if TYPE_CHECKING:
 
 
 class File:
-    def __init__(self, path: str, content: str):
+    def __init__(self, path: str, content: str) -> None:
         self.path = path
         self.content = content
 
@@ -26,10 +26,7 @@ class File:
 def parse_test_files(test_files: List[Dict[str, Any]]) -> List[File]:
     files: List[File] = []
     for test_file in test_files:
-        path = test_file.get("path")
-        if not path:
-            path = "main.py"
-
+        path = test_file.get("path", "main.py")
         file = File(path=path, content=test_file.get("content", ""))
         files.append(file)
     return files

--- a/pytest_mypy_plugins/item.py
+++ b/pytest_mypy_plugins/item.py
@@ -175,10 +175,22 @@ class YamlTestItem(pytest.Item):
         assert mypy_executable is not None
 
         # add current directory to path
-        self.environment_variables["PYTHONPATH"] = ":".join([os.environ.get("PYTHONPATH", ""), str(execution_path),])
-        self.environment_variables["MYPYPATH"] = ":".join(
-            [os.environ.get("MYPYPATH", ""), os.path.dirname(self.base_ini_fpath),]
+        self.environment_variables["PYTHONPATH"] = ":".join(
+            [
+                os.environ.get("PYTHONPATH", ""),
+                str(execution_path),
+            ]
         )
+
+        # adding proper MYPYPATH variable
+        mypy_path_parts = []
+        existing_mypy_path = os.environ.get("MYPYPATH", "")
+        if existing_mypy_path:
+            mypy_path_parts.append(existing_mypy_path)
+        if self.base_ini_fpath:
+            mypy_path_parts.append(os.path.dirname(self.base_ini_fpath))
+        self.environment_variables["MYPYPATH"] = ":".join(mypy_path_parts)
+
         # Windows requires this to be set, otherwise the interpreter crashes
         if "SYSTEMROOT" in os.environ:
             self.environment_variables["SYSTEMROOT"] = os.environ["SYSTEMROOT"]

--- a/pytest_mypy_plugins/item.py
+++ b/pytest_mypy_plugins/item.py
@@ -175,7 +175,10 @@ class YamlTestItem(pytest.Item):
         assert mypy_executable is not None
 
         # add current directory to path
-        self.environment_variables["PYTHONPATH"] = str(execution_path)
+        self.environment_variables["PYTHONPATH"] = ":".join([os.environ.get("PYTHONPATH", ""), str(execution_path),])
+        self.environment_variables["MYPYPATH"] = ":".join(
+            [os.environ.get("MYPYPATH", ""), os.path.dirname(self.base_ini_fpath),]
+        )
         # Windows requires this to be set, otherwise the interpreter crashes
         if "SYSTEMROOT" in os.environ:
             self.environment_variables["SYSTEMROOT"] = os.environ["SYSTEMROOT"]

--- a/pytest_mypy_plugins/item.py
+++ b/pytest_mypy_plugins/item.py
@@ -172,7 +172,7 @@ class YamlTestItem(pytest.Item):
         import distutils.spawn
 
         mypy_executable = distutils.spawn.find_executable("mypy")
-        assert mypy_executable is not None
+        assert mypy_executable is not None, "mypy executable is not found"
 
         # add current directory to path
         self.environment_variables["PYTHONPATH"] = ":".join(
@@ -194,6 +194,7 @@ class YamlTestItem(pytest.Item):
         # Windows requires this to be set, otherwise the interpreter crashes
         if "SYSTEMROOT" in os.environ:
             self.environment_variables["SYSTEMROOT"] = os.environ["SYSTEMROOT"]
+
         completed = subprocess.run(
             [mypy_executable, *mypy_cmd_options],
             stdout=subprocess.PIPE,

--- a/pytest_mypy_plugins/utils.py
+++ b/pytest_mypy_plugins/utils.py
@@ -117,7 +117,6 @@ def _add_aligned_message(s1: str, s2: str, error_message: str) -> str:
     maxw = 72  # Maximum number of characters shown
 
     error_message += "Alignment of first line difference:\n"
-    # sys.stderr.write('Alignment of first line difference:\n')
 
     trunc = False
     while s1[:30] == s2[:30]:
@@ -136,23 +135,18 @@ def _add_aligned_message(s1: str, s2: str, error_message: str) -> str:
 
     # Write a chunk of both lines, aligned.
     error_message += "  E: {}{}\n".format(s1[:maxw], extra)
-    # sys.stderr.write('  E: {}{}\n'.format(s1[:maxw], extra))
     error_message += "  A: {}{}\n".format(s2[:maxw], extra)
-    # sys.stderr.write('  A: {}{}\n'.format(s2[:maxw], extra))
     # Write an indicator character under the different columns.
     error_message += "     "
     # sys.stderr.write('     ')
     for j in range(min(maxw, max(len(s1), len(s2)))):
         if s1[j : j + 1] != s2[j : j + 1]:
             error_message += "^"
-            # sys.stderr.write('^')  # Difference
             break
         else:
             error_message += " "
-            # sys.stderr.write(' ')  # Equal
     error_message += "\n"
     return error_message
-    # sys.stderr.write('\n')
 
 
 def remove_empty_lines(lines: List[str]) -> List[str]:

